packages/client/src/workflow-options.ts

```diff
@@ -66,6 +66,12 @@ export function compileWorkflowOptions<T extends WorkflowOptions>(options: T): W
 }
 
 export interface WorkflowUpdateOptions {
+  /**
+   * The Update Id, which is a per-Workflow-Execution idempotency key.
```

Comment:
Perhaps we also need to say something about what the expectations and behavior are regarding uniqueness of the id w.r.t other updates on this WFE.

Reply:
```suggestion
   * The Update Id, which is a unique-per-Workflow-Execution identifier for this Update.
```

> Perhaps we also need to say something about what the expectations and behavior are regarding uniqueness of the id w.r.t other updates on this WFE.

Yeah, like if start/executeUpdate is called with an `updateId` that the Server has received before, [an X error will be thrown] or [the original Update handle/result will be returned].

Reply:
Clarified with:

```
   * The Update Id, which is a unique-per-Workflow-Execution identifier for this Update.
   *
   * We recommend setting it to a meaningful business ID or idempotency key (like a request ID) passed from upstream. If
   * it is not provided, it will be set to a random string by the Client. If the Server receives two Updates with the
   * same Update Id to a Workflow Execution with the same Run Id, the second Update will return a handle to the first
   * Update.
```